### DISCUSSION TITLE
Can the 99-unsorted.sh file pose a security threat?

### DISCUSSION THREAD

**Discussion by akim-13** [CyberShadow/aconfmgr#149] [Q&A] [answered]
I'd like to track the `~/.config/aconfmgr` directory using git by adding it to my public `dotfiles` repo. However, I'm a bit hesitant to do so as the `99-unsorted.sh` file is 10k lines long, and I'm not 100% certain what each one represents. I suppose that excluding the `files` directory from git repo would render the `*.sh` files useless to anybody except for me, but it never hurts to double-check.

**Comment by CyberShadow** [comment, marked as answer]
Have a look at what the README says about `99-unsorted.sh`. Generally, you are expected to sort its contents into multiple other files which define and explain the system configuration.

**Reply by akim-13** [reply]
Yea, I've read about this, but it seems like a big task which requires a lot of time, mainly because I'm not entirely familiar with the purpose of some packages, directories, etc. So I just wanted to make sure that these `*.sh` files are harmless without the `files` directory and wouldn't pose any security threats in a public repo. Is that corrent?

Edit: even if it is correct, I don't see a reason why other people would want to see my configuration, so I'll just create a private repo instead. Thank you.

**Reply by MenacingPerson** [reply]
If you don't know what a package does, simply remove it after finding out what it does.

Also removing the files dir will break aconfmgr, unless you can put it back

**Reply by CyberShadow** [reply]
> Yea, I've read about this, but it seems like a big task which requires a lot of time, mainly because I'm not entirely familiar with the purpose of some packages, directories, etc.

Arch Linux is attractive to users who exactly want to know what every package on their system does :)

It is indeed a big task to sort everything if you already have an extensively configured system. You may find doing this worthwhile anyway. In any case, you will likely find packages or changes on your system which you don't need any more, which will reduce the entropy (complexity) of your system, and make maintenance easier, whether or not you continue using aconfmgr.
